src/main/java/com/bettercloud/vault/rest/Rest.java

```diff
@@ -431,7 +450,15 @@ private URLConnection initURLConnection(final String urlString, final String met
         URLConnection connection = null;
         try {
             final URL url = new URL(urlString);
-            connection = url.openConnection();
+            if (proxy == null) {
+                connection = url.openConnection();
+            } else {
+                connection = url.openConnection(proxy);
+                if (proxyAuth != null) {
+                    connection.setRequestProperty("Proxy-Authenticate", proxyAuth);
```

Comment:
The JRE will disable trying to use `java.net.Authenticator#getPasswordAuthentication()` for the proxy authentication if this header is set.